### PR TITLE
Replace `_LIBCUDACXX_CPO_ACCESSIBILITY` with `_CCCL_GLOBAL_VARIABLE`

### PR DESCRIPTION
**Pull request by miscco**
The name `_LIBCUDACXX_CPO_ACCESSIBILITY` has always been less than optimal.

Replace it with a better name and make it available for all cccl libraries

### CODE REVIEW

File: libcudacxx/include/cuda/std/__cccl/dialect.h [hunk 1]
```diff
@@ -90,4 +90,25 @@
...
+#else // ^^^ _CCCL_CUDACC_BELOW_11_3 ^^^ / vvv !_CCCL_CUDACC_BELOW_11_3 vvv
+#  define _CCCL_CONSTEXPR_GLOBAL constexpr
+#endif // !_CCCL_CUDACC_BELOW_11_3
+
+// Inline variables are only available from C++17 onwards
+#if _CCCL_STD_VER >= 2017 && defined(__cpp_inline_variables) && (__cpp_inline_variables >= 201606L)
+#  define _CCCL_INLINE_VAR inline
+#else // ^^^ C++14 ^^^ / vvv C++17 vvv
+#  define _CCCL_INLINE_VAR
+#endif // _CCCL_STD_VER <= 2014
```
Comment (bernhardmgruber) [inline libcudacxx/include/cuda/std/__cccl/dialect.h:105]: Question: wouldn't this suffice:
```suggestion
#ifdef __cpp_inline_variables
#  define _CCCL_INLINE_VAR inline
#else
#  define _CCCL_INLINE_VAR
#endif
```
I know feature testing macros are officially part of C++20, but available a lot earler in many compilers. Also, there should be no need to test the feature version or the C++ version, just whether the compiler understands inline variables.

File: libcudacxx/include/cuda/std/__cccl/dialect.h [hunk 2]
```diff
@@ -90,4 +90,25 @@
...
+
+// We need to treat host and device separately
+#if defined(__CUDA_ARCH__)
+#  define _CCCL_GLOBAL_VARIABLE _CCCL_DEVICE _CCCL_CONSTEXPR_GLOBAL
+#else // ^^^ __CUDA_ARCH__ ^^^ / vvv !__CUDA_ARCH__ vvv
+#  define _CCCL_GLOBAL_VARIABLE _CCCL_INLINE_VAR constexpr
+#endif // __CUDA_ARCH__
```
Comment (bernhardmgruber) [inline libcudacxx/include/cuda/std/__cccl/dialect.h:112]: Question: why can't we make the device global variable `_CCCL_INLINE_VAR` as well? And why doesn't `_CCCL_INLINE_VAR _CCCL_CONSTEXPR_GLOBAL` suffice for both host and device?

Comment (miscco) [reply libcudacxx/include/cuda/std/__cccl/dialect.h:112]: The issue is that nvcc does not [accept inline constexpr variables:](https://godbolt.org/z/MnoM6zs99)

The only thing that we could do is mark them `static inline` but meh

Comment (bernhardmgruber) [reply libcudacxx/include/cuda/std/__cccl/dialect.h:112]: I see. Thx!

**Review by bernhardmgruber — APPROVED** [review]

File: libcudacxx/include/cuda/std/__cccl/dialect.h (hunk 1 above)
Comment (miscco) [reply libcudacxx/include/cuda/std/__cccl/dialect.h:105]: I believe older gcc compilers do not properly support inline variables despite defining the feature test macros

Comment (bernhardmgruber) [reply libcudacxx/include/cuda/std/__cccl/dialect.h:105]: Alright. Consider adding a comment on this :)

File: libcudacxx/include/cuda/std/__cccl/dialect.h (hunk 2 above)
Comment (bernhardmgruber) [inline libcudacxx/include/cuda/std/__cccl/dialect.h:112]: I was just looking for a similar macro and remembered this PR. Then I noticed `_CCCL_GLOBAL_VARIABLE` may suggest the variable to be mutable.

Suggestion: rename the macro to `_CCCL_GLOBAL_CONSTANT` to make clear it's `constexpr`.

**Review by bernhardmgruber — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**